invenio_pages/views.py

```diff
@@ -93,11 +103,13 @@ def handle_not_found(exception, **extra):
         return exception
 
 
-def _add_url_rule(url_or_urls):
+def _add_url_rule(url):
     """Register URL rule to application URL map."""
-    if isinstance(url_or_urls, str):
-        url_or_urls = [url_or_urls]
-    map(
-        lambda url: current_app.add_url_rule(url, "invenio_pages.view", view),
-        url_or_urls,
+    rule = current_app.url_rule_class(
+        url,
+        endpoint="invenio_pages.view",
+        methods=["GET", "HEAD", "OPTIONS"],
+        strict_slashes=True,
+        merge_slashes=True,
     )
+    current_app.url_map.add(rule)
```

Comment:
Comment/shelve: Since we are still in setup mode at this point, isn't it possible to use `current_app.add_url_rule` here like before? I suspect it's because of [this check](https://github.com/pallets/flask/blob/9da947a279d5ed5958609e0b6ec690160c496480/src/flask/app.py#L1359-L1365). Which is a valid check in the sense that if you did `url_for("invenio_pages.view")`, you would probably get a random URL back?

In any case, I think the best would be to have unique human-friendly endpoint names for each page rule and point them all to the `invenio_pages.views.view` function. But this unique key is not currently part of the `Page` model, so that would mean opening a can of worms and updating the alembic recipes, etc. 

The above also means that there's no way to do something like `url_for("about_page")` to get the URL for a page.

So maybe we just open an issue and revisit in the future to add this as a feature.

Reply:
We are not only in the setup mode, this same function is also used by the 404 handler to register the URLs (see code [here](https://github.com/inveniosoftware/invenio-pages/blob/628595c0cb9f7ac57706147ac9ac0e7832faa802/invenio_pages/views.py#L95)). In this situation is the [`@setupmethod`](https://github.com/pallets/flask/blob/9da947a279d5ed5958609e0b6ec690160c496480/src/flask/app.py#L1308) decorator who prevents us from using it.

However you are right, I didn't think about the issues of having all of them registered under the same endpoint name .... thanks a lot for pointing this out! 

I have created an [issue for this](https://github.com/inveniosoftware/invenio-pages/issues/61) (feel free to improve it).